streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -1846,6 +2115,23 @@ object ZStream extends ZStreamPlatformSpecific {
         )
     }
 
+  /**
+   * Creates a stream from a [[zio.ZQueue]] of values
+   * The queue will be shut down once the stream is finished
+   */
+  final def fromQueue0[R, E, A](queue: ZQueue[_, _, R, E, _, A]): ZStream[R, E, A] =
```

Comment:
I think this is `ZStream.fromInputStreamManaged(fromQueue(queue).process.ensuringFirst(queue.shutdown))`.

Reply:
And we can call it `fromQueueWithShutdown` I suppose, as it isn't a "more powerful" version of `toQueue"

Reply:
Removed as I don't think this make sense with the new process encoding.
The idea was to basically eagerly signal upstream that we are done with the queue. This was possible in fold because of cont, but I don't think it makes sense otherwise